### ISSUE TITLE
read_csv does not set index name on empty dataframe

### ISSUE THREAD

**Issue by corr723** [pydata/pandas#10184]
If foo.csv contains
```csv
x,y
1,2
```
then you get the normal behavior:
```py
>>> df = pd.read_csv('foo.csv', index_col=0)
>>> list(df.columns)
['y']
>>> df.index.name
'x'
```
but if you remove the data row "1,2" in the CSV the index name gets dropped:
```py
>>> df = pd.read_csv('foo.csv', index_col=0)
>>> list(df.columns)
['y']
>>> df.index.name
None
```
This means round-tripping with to_csv/read_csv fails for empty dataframes.

**Comment by jreback** [comment]
ok, will mark as a bug. want to do a pull-request to fix?

**Comment by corr723** [comment]
No, I don't have a patch.

**Comment by jreback** [comment]
@corr723 well if you would like to work on one would be gr8

**Comment by corr723** [comment]
No plan to work on one, sorry

**Comment by jorisvandenbossche** [comment]
Closed by #10302
